app/Http/Controllers/GroupController.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Judite\Models\Group;
+use App\Judite\Models\Course;
+use App\Judite\Models\Student;
+use App\Judite\Models\Invitation;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Auth;
+use App\Exceptions\UserHasAlreadyGroupInCourseException;
+
+class GroupController extends Controller
+{
+    /**
+     * Create a new controller instance.
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $this->middleware('can.student');
+        $this->middleware('student.verified');
+        $this->middleware('can.group');
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\View\View
+     */
+    public function index()
```

Comment:
This method really needs a refactor. From where I see it this is what this method needs:
- Get the courses which the authenticated student has enrolled. ( Student::enrolledCourses )
- For each course get only available groups, right? ( Course::availableGroups )
- For each group get the number of invitations, that is being correctly done.

What do you think?

Reply:
We should discuss this personally